### PR TITLE
Issue-2916- Provide more info about file/classes in code snippets to adding-attributes.html

### LINKED ISSUE #2916: Provide more info about file/classes in code snippets to adding-attributes.html
# Topic clarification/correction request

## Topic Link

https://devdocs.magento.com/guides/v2.2/extension-dev-guide/extension_attributes/adding-attributes.html

## Description

In the [Extension Attributes Configuration section](https://devdocs.magento.com/guides/v2.2/extension-dev-guide/extension_attributes/adding-attributes.html#extension-attributes-configuration) of [Adding extension attributes to entity page](https://devdocs.magento.com/guides/v2.2/extension-dev-guide/extension_attributes/adding-attributes.html), add more information about files and classes relevant to the provided code snippets so that it's clearer and more descriptive.

## Additional information

This is an issue/question related to customer feedback on the DevDocs site.

### PR DESCRIPTION
**Pull request by dineshvb**
## Purpose of this pull request

This pull request (PR) fixes #2916 

## Affected DevDocs pages

-  guides/v2.3/extension-dev-guide/extension_attributes/adding-attributes.html
-  guides/v2.4/extension-dev-guide/extension_attributes/adding-attributes.html

### CODE REVIEW

File: src/guides/v2.3/extension-dev-guide/extension_attributes/adding-attributes.md [hunk 1]
```diff
@@ -186,6 +188,8 @@ For scalar attributes we can use next configuration:
 </config>
 ```
 
+Here, the scalar attributes indicate the simple form of attribute representation such as an Integer or a String. The class or interface for which the extension attributes must be added is specified inside the "for" attribute of the `<extension_attributes>` tag. In this case, it is the ProductInterface. The attribute is specified with an unique code and it's type.
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/extension_attributes/adding-attributes.md:191]: (a unique is correct as unique starts with a consonant sound)
```suggestion
Here, the scalar attributes indicate the simple form of attribute representation such as an Integer or a String. The class or interface for which the extension attributes must be added is specified inside the "for" attribute of the `<extension_attributes>` tag. In this case, it is the ProductInterface. The attribute is specified with a unique code and its type.
```

File: src/guides/v2.3/extension-dev-guide/extension_attributes/adding-attributes.md [hunk 2]
```diff
@@ -206,6 +212,8 @@ For array extension attributes:
 </config>
 ```
 
+The array extension attributes are just an extension of the scalar attributes where a range of values can be represented as an attribute. `[]` symbol is used to indicate the type of the attribute as array.
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/extension_attributes/adding-attributes.md:215]:
```suggestion
The array extension attributes are just an extension of the scalar attributes where a range of values can be represented as an attribute. The `[]` symbol is used to indicate the type of the attribute as an array.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
The issue indicates there are a lot more problems with the concerned page than are covered by this PR - will you be adding more? What's the plan?

**Comment by dineshvb** [comment]
Hi @BarnyShergold , I went through the https://devdocs.magento.com/guides/v2.4/extension-dev-guide/extension_attributes/adding-attributes.html page and after I got an understanding of the issue specified, I made these changes. I don't know how to proceed hereafter. I would like to have some suggestions. Can you assist me with this?

**Comment by BarnyShergold** [comment]
@shrielenee - Some additional information to the section on your issue has been added by @dineshvb - Textually the additions are fine, can you take a look and feedback as to if there is enough new information or what might else be needed to clarify the examples as indicated in you report?

File: src/guides/v2.3/extension-dev-guide/extension_attributes/adding-attributes.md [hunk 3]
```diff
@@ -186,6 +188,8 @@ For scalar attributes we can use next configuration:
 </config>
 ```
 
+Here, the scalar attributes indicate the simple form of attribute representation such as an Integer or a String. The class or interface for which the extension attributes must be added is specified inside the "for" attribute of the `<extension_attributes>` tag. In this case, it is the ProductInterface. The attribute is specified with a unique code and its type.
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/extension_attributes/adding-attributes.md:191]:
```suggestion
Here, the scalar attributes indicate the simple form of attribute representation, such as an integer or a string. Specify the class or interface of the extension attributes inside the "for" attribute of the `<extension_attributes>` tag. In this case, it is the ProductInterface. The attribute is specified with a unique code and its type.
```

File: src/guides/v2.3/extension-dev-guide/extension_attributes/adding-attributes.md [hunk 4]
```diff
@@ -206,6 +212,8 @@ For array extension attributes:
 </config>
 ```
 
+The array extension attributes are just an extension of the scalar attributes where a range of values can be represented as an attribute. The `[]` symbol is used to indicate the type of the attribute as an array.
```
Comment (keharper) [inline src/guides/v2.3/extension-dev-guide/extension_attributes/adding-attributes.md:215]:
```suggestion
The array extension attributes are just an extension of the scalar attributes where a range of values can be represented as an attribute. The `[]` symbol indicates the attribute type is an array.
```

**Review by keharper — CHANGES_REQUESTED** [review]
Some grammatical changes.

**Comment by dineshvb** [comment]
@keharper @BarnyShergold @dobooth Changed the base to master

**Comment by shrielenee** [comment]
Agree with all @keharper comments. Thanks y'all!

**Comment by dineshvb** [comment]
@shrielenee As @BarnyShergold said, I would like feedback on my changes so that I could proceed further

**Comment by BarnyShergold** [comment]
Looks good so far.

**Review by BarnyShergold — APPROVED** [review]

**Comment by dineshvb** [comment]
@keharper @dobooth Is it good to merge ?

**Review by keharper — APPROVED** [review]

**Comment by keharper** [comment]
running tests

**Pull request merged**